### PR TITLE
test(e2e): validate that pulling from an acr registry attached to aks via --attach-acr with karpenter nodes works

### PR DESCRIPTION
**Pull request by Bryce-Soghigian**
Fixes # 

**Description**
As part of the efforts to fix 1.30 bootstrap we removed a flag that removes functionality for out of tree providers. One of the side affects of this is that ACR Pull will not work in the same way. We needed a way to validate this image pull works in a reproducible and consistent way. 

This test suite is that. Currently it has to be ran locally to target specific k8s versions so a nice followup at lower priority would be to add the ability to run our e2es against a specific k8s version. 
**How was this change tested?**
- make az-e2etests

```
[251.679 seconds]
------------------------------

Ran 1 of 1 Specs in 251.874 seconds
SUCCESS! -- 1 Passed | 0 Failed | 0 Pending | 0 Skipped
--- PASS: TestAcr (251.87s)
PASS
ok      github.com/Azure/karpenter-provider-azure/test/suites/acr       252.588s
```

**Does this change impact docs?**
- [ ] Yes, PR includes docs updates
- [ ] Yes, issue opened: # 
- [ ] No

**Release Note**

```release-note

```

### CODE REVIEW

File: Makefile-az.mk [hunk 1]
```diff
@@ -327,3 +330,12 @@ az-helm-install-snapshot: az-configure-values ## Install Karpenter snapshot rele
 
 az-rmcrds: ## Delete Karpenter CRDs
 	kubectl delete crd nodepools.karpenter.sh nodeclaims.karpenter.sh aksnodeclasses.karpenter.azure.com
+
+az-cleanenv:
```
[the remark this answers is not in the document]
Comment (Bryce-Soghigian) [reply Makefile-az.mk:334]: Just nice

File: test/suites/acr/suite_test.go [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+var _ = BeforeEach(func() {
+	env.BeforeEach()
+	nodeClass = env.DefaultAKSNodeClass()
+	nodePool = env.DefaultNodePool(nodeClass)
+})
+var _ = AfterEach(func() { env.Cleanup() })
+var _ = AfterEach(func() { env.AfterEach() })
+
+var _ = Describe("Acr", func() {
+	Describe("Image Pull", func() {
```
[the remark this answers is not in the document]
Comment (Bryce-Soghigian) [reply test/suites/acr/suite_test.go:61]: The systempool is always tainted before each of these steps in the azure/karpenter e2es see: https://github.com/Azure/karpenter-provider-azure/blob/4ec1d9709bae92a9c4a4913aa98b0421600c6438/.github/actions/e2e/install-karpenter/action.yaml#L52

File: .github/workflows/e2e-matrix.yaml [hunk 3]
```diff
@@ -43,7 +43,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        suite: [Nonbehavioral, Utilization, GPU, Drift, Integration, NodeClaim, Chaos]
+        suite: [Nonbehavioral, Utilization, GPU, Drift, Integration, NodeClaim, Chaos, Acr]
```
[the remark this answers is not in the document]
Comment (Bryce-Soghigian) [reply .github/workflows/e2e-matrix.yaml:46]: Yeah seems like a bit much

**Review by comtalyst — APPROVED** [review]

**Comment by Bryce-Soghigian** [comment]
gonna rewrite this on the azure repo rather than my fork.

**Pull request closed without merging**